### ISSUE TITLE
[Bug] `graph.neighbors_undirected(n0).detach()` returns node index 4294967295 

### ISSUE THREAD

**Issue by jerry73204** [petgraph/petgraph#501]
### Steps To Reproduce

The bug is related to `graph.neighbors_undirected(n0).detach()`, which gives a neighbor walker surrounding a specified node.
To reproduce, get the walker first. Then, remove some of the neighbors `remove_edge()` before walking through the neighbors.
In the later walking loop, it returns the node index 4294967295 if the node is removed earlier.

```rust
use petgraph::{graph::NodeIndex, stable_graph::StableUnGraph};

#[test]
fn neighbor_detach_test() {
    let mut graph = StableUnGraph::default();
    let n0: NodeIndex = graph.add_node(());
    let n1 = graph.add_node(());
    let n2 = graph.add_node(());

    let e01 = graph.add_edge(n0, n1, ());
    let e02 = graph.add_edge(n0, n2, ());
    let e12 = graph.add_edge(n1, n2, ());

    println!("Create edges:");
    println!("{}: {} -- {}", e01.index(), n0.index(), n1.index());
    println!("{}: {} -- {}", e02.index(), n0.index(), n2.index());
    println!("{}: {} -- {}", e12.index(), n1.index(), n2.index());
    println!();

    let mut neighbors = graph.neighbors_undirected(n0).detach();
    graph.remove_edge(e01);
    graph.remove_edge(e02);

    while let Some((eidx, nidx)) = neighbors.next(&graph) {
        let neighbor = graph.node_weight(nidx);
        if neighbor.is_none() {
            println!("node {} is not found", nidx.index());
        }

        let edge = graph.remove_edge(eidx);

        if edge.is_none() {
            println!("edge {} is not found", eidx.index());
        }
    }
}
```

### Actual Results

The test code prints the following:

```
Create edges:
0: 0 -- 1
1: 0 -- 2
2: 1 -- 2

node 4294967295 is not found
edge 1 is not found
node 4294967295 is not found
edge 0 is not found
```

### Expected Results

4294967295 is a special value (-1 in signed integer) indicating the node does not exist. It is expected not being returned from the walker.
